bcrypt_pbkdf.c: clang 10 warning

Suspected buffer size miscalculation:
```
clang -target i686-w64-mingw32 --sysroot /usr/i686-w64-mingw32 -fno-ident -DHAVE_STRTOI64 -DLIBSSH2_DH_GEX_NEW=1 -DHAVE_DECL_SECUREZEROMEMORY=1 -fno-asynchronous-unwind-tables -O2 -DNDEBUG -DLIBSSH2_WIN32 -DHAVE_WINDOWS_H  -fno-builtin -fno-strict-aliasing -Wall  -m32 -DLIBSSH2_OPENSSL -DLIBSSH2_HAVE_ZLIB -I../win32 -I../include -I"../../openssl/include" -I../../zlib/pkg/usr/local -c ../src/bcrypt_pbkdf.c -o release/bcrypt_pbkdf.o
Creating release/libssh2.res
../src/bcrypt_pbkdf.c:84:46: warning: expression does not compute the number of elements in this array; element type is 'uint32_t' (aka 'unsigned int'), not 'uint64_t' (aka 'unsigned long long') [-Wsizeof-array-div]
        blf_enc(&state, cdata, sizeof(cdata) / sizeof(uint64_t));
                                      ~~~~~  ^
../src/bcrypt_pbkdf.c:65:14: note: array 'cdata' declared here
    uint32_t cdata[BCRYPT_BLOCKS];
             ^
../src/bcrypt_pbkdf.c:84:46: note: place parentheses around the 'sizeof(uint64_t)' expression to silence this warning
        blf_enc(&state, cdata, sizeof(cdata) / sizeof(uint64_t));
                                             ^
1 warning generated.
```
Ref: https://ci.appveyor.com/project/curlorg/curl-for-win#L12977

**To Reproduce**
Build libssh2 with above compiler/options.

**Expected behavior**
No warning.

**Version (please complete the following information):**
 - OS: ~~Windows~~ any
 - libssh2 version: 1.9.0

--- a/src/bcrypt_pbkdf.c
+++ b/src/bcrypt_pbkdf.c
@@ -81,7 +81,7 @@ bcrypt_hash(uint8_t *sha2pass, uint8_t *sha2salt, uint8_t *out)
         cdata[i] = Blowfish_stream2word(ciphertext, sizeof(ciphertext),
                                         &j);
     for(i = 0; i < 64; i++)
-        blf_enc(&state, cdata, sizeof(cdata) / sizeof(uint64_t));
+        blf_enc(&state, cdata, BCRYPT_BLOCKS / 2);
 
     /* copy out */
     for(i = 0; i < BCRYPT_BLOCKS; i++) {

